[Improvement-11773][api] Optimize the log printing of the api module according…

guowei-su opened the pull request:
<!--Thanks very much for contributing to Apache DolphinScheduler. Please review https://dolphinscheduler.apache.org/en-us/community/development/pull-request.html before opening a pull request.-->


## Purpose of the pull request
This pull request optimizes the log printing of the api module.
<!--(For example: This pull request adds checkstyle plugin).-->

## Brief change log
- Add or modify log of the files in api/service/impl/, etc.
- close #11773
- doc [[Improvement-11481][docs] Add log specification document to contribution guidelines. #11484](https://github.com/apache/dolphinscheduler/pull/11484)
<!--*(for example:)*
  - *Add maven-checkstyle-plugin to root pom.xml*
-->
## Verify this pull request

<!--*(Please pick either of the following options)*-->

This pull request is code cleanup without any test coverage.

*(or)*

This pull request is already covered by existing tests, such as *(please describe tests)*.


<!--*(example:)*
  - *Added dolphinscheduler-dao tests for end-to-end.*
  - *Added CronUtilsTest to verify the change.*
  - *Manually verified the change by testing locally.* -->

(or)

If your pull request contain incompatible change, you should also add it to `docs/docs/en/guide/upgrede/incompatible.md`

guowei-su committed:
[DS-11773][api] Optimize the log printing of the api module according to the log specification doc.
- Resolve conflict.

SbloodyS reviewed (commented):
Please associate the issue and resolves conflicts.

guowei-su commented:
issue [#11773](https://github.com/apache/dolphinscheduler/issues/11773)

guowei-su committed:
[DS-11773][api] Optimize the log printing of the api module according to the log specification doc.
- fix ProcessDefinitionServiceImpl.java dependency

caishunfeng commented on dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/permission/ResourcePermissionCheckServiceImpl.java:
```diff
@@ -107,6 +107,8 @@ public boolean resourcePermissionCheck(Object authorizationType, Object[] needCh
             Set<?> originResSet = new HashSet<>(Arrays.asList(needChecks));
             Set<?> ownResSets = RESOURCE_LIST_MAP.get(authorizationType).listAuthorizedResource(userId, logger);
             originResSet.removeAll(ownResSets);
+            if (!originResSet.isEmpty())
```
```suggestion
            if (CollectionUtils.isNotEmpty(originResSet)) {
                 logger.warn("User does not have resource permission {} on associated resources, userId:{}", authorizationType, userId);
            }
```

caishunfeng commented on dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/TenantServiceImpl.java:
```diff
@@ -229,9 +232,14 @@ public Map<String, Object> updateTenant(User loginUser, int id, String tenantCod
-        tenantMapper.updateById(updateTenant);
-
-        putMsg(result, Status.SUCCESS);
+        int update = tenantMapper.updateById(updateTenant);
```
Need to update the UT case.

caishunfeng commented on dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/TenantServiceImpl.java:
```diff
@@ -278,10 +290,16 @@ public Map<String, Object> deleteTenantById(User loginUser, int id) throws Excep
 
-        tenantMapper.deleteById(id);
-        processInstanceMapper.updateProcessInstanceByTenantId(id, -1);
+        int delete = tenantMapper.deleteById(id);
```
Need to update the UT case.

caishunfeng commented:
Please add a doc of `How to print logs better`.

guowei-su committed:
[DS-11773][api] Optimize the log printing of the api module according to the log specification doc.
- Accept proposed change in ResourcePermissionCheckServiceImpl.java

guowei-su committed:
[DS-11773][api] Optimize the log printing of the api module according to the log specification doc.
- Update the UT case in TenantServiceTest.java, UsersServiceTest.java, TaskGroupServiceTest.java.
- Update log printing in TaskGroupServiceImpl.java.
- Fix NPE of log parameter in ExecutorServiceImpl.java when call python api.

guowei-su committed:
[DS-11773][api] Optimize the log printing of the api module according to the log specification doc.
- Fix NPE of log in WorkerGroupServiceImpl.java.
- Fix log parameter missing bug in ResourcesServiceImpl.java.

guowei-su committed:
[DS-11773][api] Optimize the log printing of the api module according to the log specification doc.
- Fix bugs and handle some vulnerability codes.

guowei-su committed:
[DS-11773][api] Optimize the log printing of the api module according to the log specification doc.
- Fix bugs and handle some vulnerability codes.

guowei-su committed:
[DS-11773][api] Optimize the log printing of the api module according to the log specification doc.
- Fix e2e User test.

guowei-su commented:
> Please add a doc of `How to print logs better`.

The doc refs this doc pr: [Add log specification document to contribution guidelines. ](https://github.com/apache/dolphinscheduler/pull/11484)

guowei-su commented:
PLAT @caishunfeng

guowei-su committed:
[DS-11773][api] Optimize the log printing of the api module according to the log specification doc.
- Fix codeQL check & SonarCloud Code Analysis.

guowei-su committed:
[DS-11773][api] Optimize the log printing of the api module according to the log specification doc.
- Resolve import conflict.

guowei-su committed:
[DS-11773][api] Optimize the log printing of the api module according to the log specification doc.
- Fix codeQL check & SonarCloud Code Analysis.

brave-lee reviewed (dismissed):
+1

brave-lee reviewed (approved):
+1

zhongjiajie commented:
I do not think this is a good idea, we should directly throw error instead of print log, and actually I already throw error in #11912

zhongjiajie commented:
> I do not think this is a good idea, we should directly throw error instead of print log, and actually I already throw error in #11912

We already have `ExceptionHandler`accept `ServiceException` and package `ServiceException` code and message into `Result` object, and it uses in the controller layer https://github.com/apache/dolphinscheduler/blob/ae6aa53f963d2ea02ab435db1e5548da23e3f882/dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/exceptions/ApiExceptionHandler.java#L39-L43
so when we throw an exception in the service layer, we will catch it in the controller layer and return the new `Result` object. And we already have exists method in https://github.com/apache/dolphinscheduler/blob/d3a77c68e62b0c471e25cd6a4bfbd2c561dd2023/dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/ProcessInstanceServiceImpl.java#L657-L659. 

when exception throw in L658 https://github.com/apache/dolphinscheduler/blob/d3a77c68e62b0c471e25cd6a4bfbd2c561dd2023/dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/ProcessInstanceServiceImpl.java#L658
controller will catch it and package into `Result(code=Status.PROCESS_INSTANCE_STATE_OPERATION_ERROR.getCode(), message=Status.PROCESS_INSTANCE_STATE_OPERATION_ERROR.getMsg())`

guowei-su commented:
> > I do not think this is a good idea, we should directly throw error instead of print log, and actually I already throw error in #11912
> 
> We already have `ExceptionHandler`accept `ServiceException` and package `ServiceException` code and message into `Result` object, and it uses in the controller layer
> 
> https://github.com/apache/dolphinscheduler/blob/ae6aa53f963d2ea02ab435db1e5548da23e3f882/dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/exceptions/ApiExceptionHandler.java#L39-L43
> 
> 
> so when we throw an exception in the service layer, we will catch it in the controller layer and return the new `Result` object. And we already have exists method in
> https://github.com/apache/dolphinscheduler/blob/d3a77c68e62b0c471e25cd6a4bfbd2c561dd2023/dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/ProcessInstanceServiceImpl.java#L657-L659
> 
> .
> when exception throw in L658
> 
> https://github.com/apache/dolphinscheduler/blob/d3a77c68e62b0c471e25cd6a4bfbd2c561dd2023/dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/ProcessInstanceServiceImpl.java#L658
> 
> 
> controller will catch it and package into `Result(code=Status.PROCESS_INSTANCE_STATE_OPERATION_ERROR.getCode(), message=Status.PROCESS_INSTANCE_STATE_OPERATION_ERROR.getMsg())`

Ok, I probably get it. I will take this approach to regulate the api module logging and exceptions in the new pr. Also, I will describe this specification in the logging specification doc.

wen-hemin commented on dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/ProjectServiceImpl.java:
```diff
@@ -538,6 +549,18 @@ public Result queryProjectCreatedAndAuthorizedByUser(User loginUser) {
+     * @param project project
+     * @return true if the user have permission to see the project, otherwise return false
+     */
+    private boolean checkReadPermission(User user, Project project) {
```
Please confirm whether it is referenced

guowei-su replied:
Sorry, this method was not referenced and was incorrectly merged while I was handling the conflict.

wen-hemin reviewed (commented):
Please confirm.
